README.md

```diff
@@ -217,5 +217,7 @@ Preview builds (aka 'nightly' builds) are distributed using https://feedz.io/ fo
 
 ## Release process
 
-Before making a new release the version in the [csproj](Consul/Consul.csproj), in the [README](README.md), in the [CI pipeline](.github/workflows/ci.yml) and in the [CHANGELOG](CHANGELOG.md) files should be updated.
-The new versions can be released using the [releases](https://github.com/G-Research/consuldotnet/releases) page.
+1. Before making a new release, make sure that the version in [Directory.Build.props](Directory.Build.props) is up to date.
+2. Then, draft a new [GitHub Release](https://github.com/G-Research/consuldotnet/releases) with a list of changes contained in the new version. This list of changes should be committed to the [CHANGELOG](CHANGELOG.md) file for posterity.
+3. When the draft release is ready, publish the it from the GitHub web UI. This will trigger GitHub to push a new tag (as specified in the new Release entry) which will build and push a new stable release to NuGet.org.
+4. If everything went well, congrats, the new versions should be live on NuGet by now. As a final post-release step, bump the version in [Directory.Build.props](Directory.Build.props) to the next unreleased version number, so that nightly build versioning can benefit.
```

Comment:
In the same step, we should probably also add the 'Unreleased' entry to the changelog?

Reply:
Yeah, I wasn't sure what to do with it when I published the new release. My guess is that it is supposed to be updated with each new PR and serves as a base for drafting a new release?

Reply:
I would think so, yes - let's try to request that from contributors in the future (or simply add it ourselves to their branch right before merging)